lib/fizzy/span.hpp

```diff
@@ -0,0 +1,55 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <iterator>
+#include <type_traits>
+
+namespace fizzy
+{
+/// The span describes an object that can refer to a contiguous sequence of objects with the first
+/// element of the sequence at position zero.
+///
+/// This is minimal implementation of C++20's std::span:
+/// https://en.cppreference.com/w/cpp/container/span
+/// Only `const T` is supported.
+template <typename T, typename = typename std::enable_if_t<std::is_const_v<T>>>
+class span
+{
+    T* const m_begin = nullptr;
```

Comment:
Did you remove first `const` here because `T` is already `const`?

Reply:
Yes, but I'm trying to require `T` to be `const X` only by the `enable_if`. The C++20 can handle also mutable spans.

I will do one more round of constness correction. And some `constexpr` seem to be missing. But the goal is the have signatures of all methods be exactly as in `std::span`. E.g. `constexpr reverse_iterator rbegin() const noexcept;`.